pkg/registry/registry.go

```diff
@@ -368,29 +369,24 @@ func (a *appsList) fetch(r *registryFetchState, fetchAll bool) error {
 	return nil
 }
 
+func ParseSlug(raw json.RawMessage) string {
+	var slug string
+	if err := json.Unmarshal(raw, &slug); err != nil {
+		return ""
+	}
+	return slug
+}
+
 func (a *appsList) Paginated(sortBy string, reverse bool, limit int) *AppsPaginated {
-	sortBySlug := sortBy == "slug"
 	sort.Slice(a.list, func(i, j int) bool {
-		vi := a.list[i]
-		vj := a.list[j]
-		var less, equal bool
-		switch valA := vi[sortBy].(type) {
-		case string:
-			valB := vj[sortBy].(string)
-			less = valA < valB
-			if !sortBySlug && !less {
-				equal = valA == valB
-			}
-		case int:
-			valB := vj[sortBy].(int)
-			less = valA < valB
-			if !sortBySlug && !less {
-				equal = valA == valB
-			}
-		}
+		valA := a.list[i][sortBy]
+		valB := a.list[j][sortBy]
+		cmp := bytes.Compare([]byte(valA), []byte(valB))
+		equal := cmp == 0
```

Comment:
Since this `equal` variable is only used to modify the value of `less` based on the second comparison of slugs, I would suggest not assigning `cmp == 0` to it and do the comparison directly in the next `if`.

Reply:
The `equal` and `less` variables are here to make the code more readable, but I agree that they could be removed. I have a light preference for keeping them, but if you prefer that we remove them and just use `cmp`, I can do that. What do you think?